src/Views/PartitioningView.vala

```diff
@@ -0,0 +1,302 @@
+        description.max_width_chars = 72;
+        description.use_markup = true;
+        description.wrap = true;
+
+        disk_list = new Gtk.Box (Gtk.Orientation.VERTICAL, 6);
+        disk_list.valign = Gtk.Align.START;
+        disk_list.margin = 6;
+        disk_list.margin_end = 12;
+
+        var disk_scroller = new Gtk.ScrolledWindow (null, null);
+        disk_scroller.hexpand = true;
+        disk_scroller.hscrollbar_policy = Gtk.PolicyType.NEVER;
+        disk_scroller.add (disk_list);
+
+        content_area.attach (disk_scroller, 0, 0);
+        content_area.attach (description, 0, 1);
+
+        load_disks ();
+
+        gparted_button = new Gtk.Button.with_label (_("Modify Partitions…"));
+        gparted_button.clicked.connect (() => open_gparted ());
+        action_area.add (gparted_button);
+        action_area.set_child_secondary (gparted_button, true);
+        action_area.set_child_non_homogeneous (gparted_button, true);
+
+        next_button = new Gtk.Button.with_label (_("Erase and Install"));
+        next_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+        next_button.sensitive = false;
+        next_button.clicked.connect (() => next_step ());
+        action_area.add (next_button);
+
+        show_all ();
+    }
+
+    private void load_disks () {
+        disks = Distinst.Disks.probe ();
+        disks.initialize_volume_groups ();
+        label_sizer = new Gtk.SizeGroup (Gtk.SizeGroupMode.BOTH);
+
+        foreach (unowned Distinst.Disk disk in disks.list ()) {
+            // Skip root disk or live disk
+            if (disk.contains_mount ("/") || disk.contains_mount ("/cdrom")) {
+                continue;
+            }
+
+            var sector_size = disk.get_sector_size ();
+            var size = disk.get_sectors () * sector_size;
+
+            string path = Utils.string_from_utf8 (disk.get_device_path ());
+
+            string model = disk.get_model ();
+            string label = (model.length == 0)
+                ? disk.get_serial ().replace ("_", " ")
+                : model;
+
+            var partitions = new Gee.ArrayList<PartitionBar> ();
+            foreach (unowned Distinst.Partition part in disk.list_partitions ()) {
+                var partition = new PartitionBar (part, path, sector_size, false, this.set_mount, this.unset_mount, this.mount_is_set, this.decrypt);
+                partitions.add (partition);
+            }
+
+            var disk_bar = new DiskBar (model, path, size, (owned) partitions);
+            label_sizer.add_widget (disk_bar.label);
+            disk_list.pack_start (disk_bar);
+        }
+
+        foreach (unowned Distinst.LvmDevice disk in disks.list_logical ()) {
+            add_logical_disk (disk);
+        }
+
+        disk_list.show_all ();
+    }
+
+    private void open_gparted () {
+        try {
+            var process = new GLib.Subprocess.newv ({"gparted"}, GLib.SubprocessFlags.NONE);
```

Comment:
Giving that gparted is providing a .desktop file, I really prefer the use of https://valadoc.org/gio-2.0/GLib.AppInfo.html

Reply:
@tintou does that let us still wait and refresh the partitions when it exits?